src/System.Text.Http.Parser/HttpUtilities.cs

```diff
@@ -0,0 +1,330 @@
+        private readonly static ulong _httpOptionsMethodLong = GetAsciiStringAsLong("OPTIONS ");
+        private readonly static ulong _httpTraceMethodLong = GetAsciiStringAsLong("TRACE \0\0");
+
+        private const ulong _http10VersionLong = 3471766442030158920; // GetAsciiStringAsLong("HTTP/1.0"); const results in better codegen
+        private const ulong _http11VersionLong = 3543824036068086856; // GetAsciiStringAsLong("HTTP/1.1"); const results in better codegen
+
+        private readonly static ulong _mask8Chars = GetMaskAsLong(new byte[] { 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff });
+        private readonly static ulong _mask7Chars = GetMaskAsLong(new byte[] { 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0x00 });
+        private readonly static ulong _mask6Chars = GetMaskAsLong(new byte[] { 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0x00, 0x00 });
+        private readonly static ulong _mask5Chars = GetMaskAsLong(new byte[] { 0xff, 0xff, 0xff, 0xff, 0xff, 0x00, 0x00, 0x00 });
+        private readonly static ulong _mask4Chars = GetMaskAsLong(new byte[] { 0xff, 0xff, 0xff, 0xff, 0x00, 0x00, 0x00, 0x00 });
+
+        private readonly static Tuple<ulong, ulong, HttpMethod, int>[] _knownMethods =
+        {
+            Tuple.Create(_mask4Chars, _httpPutMethodLong, HttpMethod.Put, 3),
+            Tuple.Create(_mask5Chars, _httpPostMethodLong, HttpMethod.Post, 4),
+            Tuple.Create(_mask5Chars, _httpHeadMethodLong, HttpMethod.Head, 4),
+            Tuple.Create(_mask6Chars, _httpTraceMethodLong, HttpMethod.Trace, 5),
+            Tuple.Create(_mask6Chars, _httpPatchMethodLong, HttpMethod.Patch, 5),
+            Tuple.Create(_mask7Chars, _httpDeleteMethodLong, HttpMethod.Delete, 6),
+            Tuple.Create(_mask8Chars, _httpConnectMethodLong, HttpMethod.Connect, 7),
+            Tuple.Create(_mask8Chars, _httpOptionsMethodLong, HttpMethod.Options, 7),
+        };
+
+        private readonly static string[] _methodNames = CreateMethodNames();
+
+        private static string[] CreateMethodNames()
+        {
+            var methodNames = new string[9];
+            methodNames[(byte)HttpMethod.Get] = "GET";
+            methodNames[(byte)HttpMethod.Put] = "PUT";
+            methodNames[(byte)HttpMethod.Delete] = "DELETE";
+            methodNames[(byte)HttpMethod.Post] = "POST";
+            methodNames[(byte)HttpMethod.Head] = "HEAD";
+            methodNames[(byte)HttpMethod.Trace] = "TRACE";
+            methodNames[(byte)HttpMethod.Patch] = "PATCH";
+            methodNames[(byte)HttpMethod.Connect] = "CONNECT";
+            methodNames[(byte)HttpMethod.Options] = "OPTIONS";
+            return methodNames;
+        }
+
+        private unsafe static ulong GetAsciiStringAsLong(string str)
+        {
+            Debug.Assert(str.Length == 8, "String must be exactly 8 (ASCII) characters long.");
+
+            Span<byte> buffer = new byte[8];
```

Comment:
This method is called once to initialize statics, but I will add a TODO to optimize it.